samples/javascript_nodejs/52.teams-messaging-extensions-search-auth-config/bots/teamsMessagingExtensionsSearchAuthConfigBot.js

```diff
@@ -32,11 +39,74 @@ class TeamsMessagingExtensionsSearchAuthConfigBot extends TeamsActivityHandler {
         // Save state changes
         await this.userState.saveChanges(context);
     }
+    async handleTeamsAppBasedLinkQuery(context, query) {
+        const magicCode =
+            query.state && Number.isInteger(Number(query.state))
+                ? query.state
+                : '';
+        const tokenResponse = await context.adapter.getUserToken(
+            context,
+            this.connectionName,
+            magicCode
+        );
 
-    async handleTeamsMessagingExtensionConfigurationQuerySettingUrl(context, query) {
+        if (!tokenResponse || !tokenResponse.token) {
+            // There is no token, so the user has not signed in yet.
+
+            // Retrieve the OAuth Sign in Link to use in the MessagingExtensionResult Suggested Actions
+            const signInLink = await context.adapter.getSignInLink(
+                context,
+                this.connectionName
+            );
+
+            return {
+                composeExtension: {
+                    type: 'auth',
+                    suggestedActions: {
+                        actions: [
+                            {
+                                type: 'openUrl',
+                                value: signInLink,
+                                title: 'Bot Service OAuth'
+                            },
+                        ],
+                    },
+                },
+            };
+        }
+        const graphClient = new SimpleGraphClient(tokenResponse.token);
+        const profile = await graphClient.GetMyProfile();
+        const attachment = CardFactory.thumbnailCard(
+            'Thumbnail Card',
+            profile.Displayname,
```

Comment:
```suggestion
            profile.displayName,
```

I believe the casing needs changing, based on the [Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer). Please test and confirm.

Reply:
Resolved